### PR TITLE
Support Cairo 0.10.3

### LINKED ISSUE #553: Update to cairo-lang 0.10.3
### Feature Request

Pre-release of cairo-lang 0.10.3 has already been relased on github, and devnet seems to be already working on integration, so we should be able to start updating on our side.

### LINKED ISSUE #554: Support TESTNET2
### Feature Request

We should support creating clients with `"testnet2"`/`TESTNET2` the same as it is done with mainnet and testnet currently.

### PR DESCRIPTION
**Pull request by war-in**
Closes #554 
Closes #553 

## Introduced changes

- StarknetChainId as our enum
- StarknetChainId.TESTNET2 added
- TESTNET2 network added

##

- [ ] This PR contains breaking changes

### CODE REVIEW

**Review by cptartur — COMMENTED** [review]
I'd make sure, that all tests/methods that accepted `TESTNET/MAINNET` also accept `TESTNET2` now but it looks fine 👍

**Review by drknzz — APPROVED** [review]

**Comment by cptartur** [comment]
Seems that devnet released support for 0.10.3 already, so I guess we can revert the reverted changes 😅

https://github.com/Shard-Labs/starknet-devnet/releases/tag/v0.4.2

File: pyproject.toml [hunk 1]
```diff
@@ -21,8 +21,8 @@ enum-tools = { extras = ["sphinx"], version = "0.9.0.post1", optional = true }
 crypto-cpp-py = "^1.0.4"
 marshmallow = "^3.15.0"
 marshmallow-oneofschema = "^3.0.1"
-cairo-lang = "0.10.2"
 typing-extensions = "^4.3.0"
+cairo-lang = "^0.10.3"
```
Comment (cptartur) [inline pyproject.toml:25]: Can we restrict it to a specific version like it was before?

Comment (war-in) [reply pyproject.toml:25]: @Solpatium I remember we had a conversation about it once. Should we change it to `0.10.3`?

Comment (war-in) [reply pyproject.toml:25]: Will change it to: https://python-poetry.org/docs/dependency-specification/#exact-requirements

File: pyproject.toml [hunk 2]
```diff
@@ -21,8 +21,8 @@ enum-tools = { extras = ["sphinx"], version = "0.9.0.post1", optional = true }
 crypto-cpp-py = "^1.0.4"
 marshmallow = "^3.15.0"
 marshmallow-oneofschema = "^3.0.1"
-cairo-lang = "0.10.2"
 typing-extensions = "^4.3.0"
+cairo-lang = "==0.10.3"
```
Comment (cptartur) [inline pyproject.toml:25]: I'm not sure if `==` is required, or it can just be `0.10.3`. But either is probably fine.

**Review by cptartur — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (war-in) [reply pyproject.toml:25]: https://python-poetry.org/docs/dependency-specification/#exact-requirements

**Pull request merged**